larq/optimizers_v2.py

```diff
@@ -58,13 +58,23 @@ def _create_slots(self, var_list):
                 self.add_slot(var, "m")
 
     def apply_gradients(self, grads_and_vars, name=None):
+        # convert from zip object so we can loop multiple times:
+        grads_and_vars = list(grads_and_vars)
+
         bin_grads_and_vars = [(g, v) for g, v in grads_and_vars if self.is_binary(v)]
         fp_grads_and_vars = [(g, v) for g, v in grads_and_vars if not self.is_binary(v)]
```

Comment:
We can also change this to:
```python
bin_grads_and_vars, fp_grads_and_vars = [], []
for grad, var in grads_and_vars:
    if self.is_binary(v):
        bin_grads_and_vars.append((grad, var))
    else:
        fp_grads_and_vars.append((grad, var))
```
Though I am not sure what's cleaner. @AdamHillier What do you think?

Reply:
Yes was unsure about this, it's a bit more verbose but I actually like this better.